File: src/ui/notes-ui/src/types/Note.ts
export interface Note {
  id: string
  title: string
  content: string
  createdAtUtc: string
  updatedAtUtc: string | null
  isActive: boolean
}

export interface CreateNoteRequest {
  title: string
  content: string
}

export interface UpdateNoteRequest {
  title: string
  content: string
}

File: src/api/NotesApi/Features/Notes/Note.cs
namespace NotesApi.Features.Notes;

public class Note
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
    public bool IsActive { get; set; }
}
